pkg/envoy/lds.go

```diff
@@ -101,3 +146,261 @@ func NewInboundListenerCheck(configGetter ConfigGetter, osmVersion osm.Controlle
+			}
+		}
+	}
+	return possibleOutboundFilterChainNames
+}
+
+func getRuleTypesFromMatchingTrafficTargetsV1alpha2(srcPod *corev1.Pod, dstPod *corev1.Pod, accessClient smiAccessClient.Interface) (map[string]struct{}, error) {
+	trafficTargets, err := accessClient.AccessV1alpha2().TrafficTargets(dstPod.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Err(err).Msgf("Error getting TrafficTargets for namespace %s", dstPod.Namespace)
+		return nil, err
+	}
+
+	ruleTypes := map[string]struct{}{}
+	for _, trafficTarget := range trafficTargets.Items {
+		if !v1alpha2.DoesTargetMatchPods(trafficTarget.Spec, srcPod, dstPod) {
+			continue
+		}
+		for _, rule := range trafficTarget.Spec.Rules {
+			ruleTypes[rule.Kind] = struct{}{}
+		}
+	}
+
+	return ruleTypes, nil
+}
+
+func getRuleTypesFromMatchingTrafficTargetsV1alpha3(srcPod *corev1.Pod, dstPod *corev1.Pod, accessClient smiAccessClient.Interface) (map[string]struct{}, error) {
+	trafficTargets, err := accessClient.AccessV1alpha3().TrafficTargets(dstPod.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Err(err).Msgf("Error getting TrafficTargets for namespace %s", dstPod.Namespace)
+		return nil, err
+	}
+
+	ruleTypes := map[string]struct{}{}
+	for _, trafficTarget := range trafficTargets.Items {
+		if !v1alpha3.DoesTargetMatchPods(trafficTarget.Spec, srcPod, dstPod) {
+			continue
+		}
+		for _, rule := range trafficTarget.Spec.Rules {
+			ruleTypes[rule.Kind] = struct{}{}
+		}
+	}
+
+	return ruleTypes, nil
+}
+
+func findMatchingFilterChainNames(envoyConfig *Config, expectedListenerName string, possibleFilterChainNames map[string]bool) error {
+	var actualFilterChainNames []string
+	var actualListeners []string
+	foundListener := false
+
+	for _, actualListener := range envoyConfig.Listeners.GetDynamicListeners() {
+		actualListeners = append(actualListeners, actualListener.Name)
+		if expectedListenerName == actualListener.Name {
+			foundListener = true
+			var listener envoy_config_listener_v3.Listener
+			activeStateListener := actualListener.GetActiveState().GetListener()
+			if activeStateListener == nil {
+				return ErrEnvoyActiveStateListenerMissing
+			}
+			if err := activeStateListener.UnmarshalTo(&listener); err != nil {
+				return ErrUnmarshalingListener
+			}
+			for _, listenerFilter := range listener.FilterChains {
+				// Check filter chain name
+				actualFilterChainNames = append(actualFilterChainNames, listenerFilter.Name)
+				listenerFilter.GetFilterChainMatch()
+				if found, exists := possibleFilterChainNames[listenerFilter.Name]; exists && !found {
+					// Found expected filter chain
+					possibleFilterChainNames[listenerFilter.Name] = true
+				}
+			}
+		}
+	}
+	if !foundListener {
+		log.Error().Msgf("must have dynamic listener with name %s but only found %s", expectedListenerName, actualListeners)
```

Comment:
The second format string should be v instead of s as per https://stackoverflow.com/questions/24489384/how-to-print-the-values-of-slices

		log.Error().Msgf("must have dynamic listener with name %s but only found %v", expectedListenerName, actualListeners)

Reply:
Good catch, thanks!